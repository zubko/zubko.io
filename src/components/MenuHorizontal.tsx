/**
 * Top bar menu component
 */

import { Fragment } from "react";
import Link from "./Link";

type Props = {
  links: { path: string; title: string }[];
};

const Separator = () => <span css={Styles.separator}> </span>;

const MenuHorizontal = ({ links, ...otherProps }: Props) => {
  return (
    <div
      css={{
        display: "flex",
        flexDirection: "row",
      }}
      {...otherProps}
    >
      {links.map((item, index) => (
        <Fragment key={item.path}>
          <Link usePlainStyle to={item.path} css={Styles.menuItem}>
            {item.title}
          </Link>
          {index < links.length - 1 ? <Separator /> : null}
        </Fragment>
      ))}
    </div>
  );
};

export default MenuHorizontal;

const Styles = {
  menuItem: {
    textDecoration: "none",
    ":hover": {},
    fontSize: "1rem",
    fontWeight: "medium",
    background: "none",
  },
  separator: {
    color: "white",
    marginLeft: "0.5rem",
    marginRight: "0.5rem",
    cursor: "default",
  },
};
