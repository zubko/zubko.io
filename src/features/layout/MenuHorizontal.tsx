/**
 * Top bar menu component
 */

import styled from "@emotion/styled";
import { Fragment } from "react";
import { Link } from "../../components/Link";

type Props = {
  links: { path: string; title: string }[];
};

export const MenuHorizontal = ({ links, ...otherProps }: Props) => {
  return (
    <div
      css={{
        display: "flex",
        flexDirection: "row"
      }}
      {...otherProps}
    >
      {links.map((item, index) => (
        <Fragment key={item.path}>
          <MenuLink usePlainStyle to={item.path}>
            {item.title}
          </MenuLink>
          {index < links.length - 1 ? <Separator /> : null}
        </Fragment>
      ))}
    </div>
  );
};

const MenuLink = styled(Link)`
  text-decoration: none;
  :hover: {
  }
  fontsize: 1rem;
  font-weight: medium;
  background: none;
`;

const Separator = styled.span`
  color: white;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  cursor: default;
`;
