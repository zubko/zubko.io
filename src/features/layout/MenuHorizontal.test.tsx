import renderer from "react-test-renderer";

import { MenuHorizontal } from "./MenuHorizontal";

describe("MenuHorizontal", () => {
  it("renders without regressions", () => {
    const tree = renderer
      .create(
        <MenuHorizontal
          links={[
            { path: "/path-1", title: "title-1" },
            { path: "/path-2", title: "title-2" }
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
