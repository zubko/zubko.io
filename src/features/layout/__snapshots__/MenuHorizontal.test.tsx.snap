// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuHorizontal renders without regressions 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-u06qw9-MenuHorizontal{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}",
      }
    }
    data-emotion="css u06qw9-MenuHorizontal"
  />,
  <div
    className="css-u06qw9-MenuHorizontal"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1be2rwe{-webkit-text-decoration:none;text-decoration:none;fontsize:1rem;font-weight:medium;background:none;}",
        }
      }
      data-emotion="css 1be2rwe"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-2w87gj-Link{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;fontsize:1rem;font-weight:medium;background:none;}.css-2w87gj-Link:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
        }
      }
      data-emotion="css 2w87gj-Link"
    />
    <a
      className="css-2w87gj-Link"
      href="/path-1"
    >
      title-1
    </a>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-11b9gp{color:white;margin-left:0.5rem;margin-right:0.5rem;cursor:default;}",
        }
      }
      data-emotion="css 11b9gp"
    />
    <span
      className="css-11b9gp"
    />
    <a
      className="css-2w87gj-Link"
      href="/path-2"
    >
      title-2
    </a>
  </div>,
]
`;
